### PR TITLE
[DDW-515] Implement translations for the application menu

### PR DESCRIPTION
**Pull request by daniloprates**
This PR implements translations for the application menu.

## Todo:

- [x] Get the user locale from the main index
- [x] Load the correct translations JSON
- [x] Reload the menus once the user changes the language on the Settings > Language 
- [x] Reload the menus once user selects the language on the initial setup Language selection screen
- [x] Japanese translation - @nikolaglumac 

## Screenshots:

image

---

### Basics

- [x] PR is updated to the most recent version of target branch (and there are no conflicts)
- [x] PR has good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [x] CHANGELOG entry has been added and is linked to the correct PR on GitHub
- [x] Automated tests: All acceptance tests are passing (`yarn run test`)
- [x] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn run dev`)
- [x] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn run package` / CI builds)
- [x] There are no *flow* errors or warnings (`yarn run flow:test`)
- [x] There are no *lint* errors or warnings (`yarn run lint`)
- [ ] Text changes are proofread and approved (Jane Wild)
- [ ] There are no missing translations (running `yarn run manage:translations` produces no changes)
- [x] UI changes look good in all themes (Alexander Rukin)
- [x] Storybook works and no stories are broken (`yarn run storybook`)
- [x] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [x] Important parts of the code are properly documented and commented
- [x] Code is properly typed with flow
- [x] React components are split-up enough to avoid unnecessary re-rendering
- [x] Any code that only works in Electron is neatly separated from components

### Testing
- [x] New feature / change is covered by acceptance tests
- [x] All existing acceptance tests are still up-to-date
- [x] New feature / change is covered by Daedalus Testing scenario
- [x] All existing Daedalus Testing scenarios are still up-to-date

### After Review:
- [ ] Merge PR
- [ ] Delete source branch
- [ ] Move ticket to `done` on the Youtrack board

### CODE REVIEW

File: source/main/index.js [hunk 1]
```diff
@@ -103,37 +64,17 @@ const onAppReady = async () => {
...
-
   mainWindow.on('close', async (event) => {
     Logger.info('mainWindow received <close> event. Safe exiting Daedalus now.');
     event.preventDefault();
     await safeExit();
   });
 
+  // TODO: Figure out how to get the locale
+  // const locale = 'en-US.json';
+  const locale = 'ja-JP.json';
```
Comment (nikolaglumac) [inline source/main/index.js]: @daniloprates you should use electron-storage to read this info: https://github.com/input-output-hk/daedalus/blob/develop/source/renderer/app/api/utils/localStorage.js#L30
You can use this call in the main process to. Just be aware that initially before user selects the language from the initial setup language-selection we should always fallback to English. That exactly what we do in the rendering process: https://github.com/input-output-hk/daedalus/blob/develop/source/renderer/app/stores/ProfileStore.js#L99-L103

Comment (nikolaglumac) [reply source/main/index.js]: @MarcusHurney FYI - read my previous comment 🙏

**Comment by MarcusHurney** [comment]
@nikolaglumac 
Getting these 4 errors from the acceptance tests, otherwise it is ready for review. Any prior knowledge as to if these have shown up recently?

screen shot 2019-01-17 at 10 20 45 pm
screen shot 2019-01-17 at 10 21 00 pm

**Comment by nikolaglumac** [comment]
@MarcusHurney all tests pass on my machine 🤔 

screen shot 2019-01-18 at 10 32 52

I have updated the cardano version Daedalus uses - please try the following:
1. Switch to `develop` branch on Cardano-sl side
2. Run tests again

File: source/renderer/app/index.js [hunk 2]
```diff
@@ -27,11 +27,10 @@ configure({
 addLocaleData([...en, ...de, ...hr, ...ja]);
 
 const environment = global.environment;
-const { NODE_ENV, NETWORK } = environment;
-const isTest = NODE_ENV === 'test';
+const { isTest, network } = environment;
 
 const initializeDaedalus = () => {
-  const api = setupApi(isTest, String(NETWORK));
+  const api = setupApi(isTest, String(network));
```
Comment (nikolaglumac) [inline source/renderer/app/index.js]: Great catch @MarcusHurney 🚀

**Comment by nikolaglumac** [comment]
@MarcusHurney we'll get the missing translations on Monday.

File: source/renderer/app/index.js (hunk 2 above)
Comment (MarcusHurney) [reply source/renderer/app/index.js]: @nikolaglumac 
I initially had the same ipc message to re-render the application menu in place for the Select Language screen but I didn't know if it was needed. I can add that in.

**Comment by MarcusHurney** [comment]
The application menu now re-renders once the user selects the language on the initial setup Language selection screen.

**Pull request merged**